trie-db/src/trie_codec.rs

```diff
@@ -111,7 +111,15 @@ impl<C: NodeCodec> EncoderStackEntry<C> {
 			self.node.node_plan()
 		};
 		let mut encoded = match node_plan {
-			NodePlan::Empty | NodePlan::Leaf { .. } => node_data.to_vec(),
+			NodePlan::Empty => node_data.to_vec(),
+			NodePlan::Leaf { partial, value } =>
+				if self.omit_value {
+					let partial = partial.build(node_data);
+					let value = value.build(node_data);
+					C::leaf_node(partial.right_iter(), partial.len(), value)
```

Comment:
Given the naming `omit_value` and then still seeing this here, I'm a little bit confused :P

Reply:
it s a 0 length value. Will rewrite that tomorrow to make it explicit.